lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
   module Adapters
     module ExternalIdentifierAdapter
       class Datacite < Abstract
-        IDENTIFIER_STATUS = { public: 'public',
-                              reserved: 'reserved',
-                              unavailable: 'unavailable' }.freeze
+        DOI_STATES = [:draft, :findable, :registered].freeze
+
+        # https://schema.datacite.org/meta/kernel-4.4/doc/DataCite-MetadataKernel_v4.4.pdf
+        REQUIRED_DATACITE_METADATA_PROPERTIES = [:creators,
+                                                 :publisher,
+                                                 :publication_year,
+                                                 :resource_type_general,
+                                                 :title,
+                                                 :url].freeze
+
+        # https://schema.datacite.org/meta/kernel-4.4/doc/DataCite-MetadataKernel_v4.4.pdf
+        SUPPORTED_DATACITE_METADATA_PROPERTIES = [:subject,
+                                                  :stuff].concat(REQUIRED_DATACITE_METADATA_PROPERTIES).freeze
+        attr_reader :rest_api
```

Comment:
Is this rest_api property accessed anywhere outside of this class?  If not, no need to make it publicly visible with `attr_reader`.

Reply:
`attr_reader` call removed.